### DISCUSSION TITLE
Add support for geometric and harmonic mean

### DISCUSSION THREAD

**Discussion by lulunac27a** [johnkerl/miller#1524] [Ideas]
I want to calculate the geometric and harmonic mean of a list of numbers. Geometric mean is the product of numbers and finds the root of the product. The harmonic mean is the reciprocal of the average of reciprocals from a list of numbers.
For example: `[3,4,5]` has geometric mean of approximately 3.91 `((3*4*5)^(1/3))` or `e^((ln(3)+ln(4)+ln(5))/3)`, and harmonic mean of approximately 3.83 `(3/(1/3+1/4+1/5))`

**Comment by johnkerl** [comment]
Super-easy to implement -- great idea @lulunac27a ! :)

**Comment by lulunac27a** [comment]
Formula for geometric mean in Go:
```go
func geometricMean(nums ...float64) {
logSum := 0
count := 0
for _, num := range nums {
logSum += math.Log(num)
count++
}
return math.Exp(logSum / count)
}
```

Formula for harmonic mean in Go:
```go
func harmonicMean(nums ...float64) {
reciprocalSum := 0
count := 0
for _, num := range nums {
reciprocalSum += 1/num
count++
}
return count / reciprocalSum
}
```
